### DISCUSSION TITLE
Middleware

### DISCUSSION THREAD

**Discussion by cdtut** [honojs/hono#2468] [Q&A]
How do make middleware run only on POST functions?

I want functions to access a value I set in middleware. Is there particular place for that or can middleware just set it anywhere on `c` like `c.value = "from middleware"`?

**Comment by NicoPlyley** [comment]
Yes, in your middleware you have full access to the context. You can use [.get()/.set()](https://hono.dev/api/context#set-get) to add variables in middleware that can be accessed by your routes.

If you want it to only run on routes using `POST`. You can add it into the route itself 
```ts
app.post('/', myCustomMiddleware, (c) => {
  // route logic
})
```

Or you can create a middleware for all routes that checks the `c.req.method` and only set the value is the method is `POST`
